configure.sh

```diff
@@ -652,7 +652,7 @@ if [ "$nls" -ne 0 ]; then
 	flags="-DBC_ENABLE_NLS=1 -DBC_ENABLED=$bc -DDC_ENABLED=$dc -DBC_ENABLE_SIGNALS=$signals"
 	flags="$flags -DBC_ENABLE_HISTORY=$hist"
 	flags="$flags -DBC_ENABLE_EXTRA_MATH=$extra_math -I./include/"
-	flags="$flags -D_POSIX_C_SOURCE=200112L -D_XOPEN_SOURCE=600"
+	flags="$flags -D_POSIX_C_SOURCE=200112L -D_XOPEN_SOURCE=600 -std=c99"
 
```

Comment:
Also, adding the non-standard `-std=c99` flag is not good. I only add options that are guaranteed/*required* to be supported by POSIX `c99`. (See the [POSIX `c99` standard page][1].)

[1]: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/c99.html#top

Reply:
Default behavior across different compilers/versions. Consistent on Solaris 10 and 11:

```
z@solaris10:~$ gcc -dM -E - </dev/null | grep STDC
#define __STDC_HOSTED__ 1
#define __STDC_UTF_16__ 1
#define __STDC_VERSION__ 201112L
#define __GNUC_STDC_INLINE__ 1
#define __STDC_UTF_32__ 1
z@solaris11:~$ gcc -dM -E - </dev/null -std=c99 | grep STDC
#define __STDC_HOSTED__ 1
#define __STDC_VERSION__ 199901L
#define __GNUC_STDC_INLINE__ 1
#define __STDC__ 1
```

Here, `__STDC__` is not always defined (e.g. on Solaris). Refer to this doc, as it may just be a GCC issue: https://gcc.gnu.org/onlinedocs/cpp/Standard-Predefined-Macros.html

on macOS for example:

```
dahlia:~ z$ gcc -dM -E - </dev/null | grep STDC
#define __GNUC_STDC_INLINE__ 1
#define __STDC_HOSTED__ 1
#define __STDC_UTF_16__ 1
#define __STDC_UTF_32__ 1
#define __STDC_VERSION__ 201112L
#define __STDC__ 1
dahlia:~ z$ gcc -dM -E - </dev/null -std=c99 | grep STDC
#define __GNUC_STDC_INLINE__ 1
#define __STDC_HOSTED__ 1
#define __STDC_UTF_16__ 1
#define __STDC_UTF_32__ 1
#define __STDC_VERSION__ 199901L
#define __STDC__ 1
```

Reply:
In this case, I have made it clear for users that they need to be building with a C99-compatible compiler. This means that the burden of adding the `-std=c99` option is on users, especially since on some platforms, there is no compiler that accepts this option. (OpenBSD is the first example that comes to mind. They build with `pcc`.) I do not want to break the build on those platforms.

Reply:
Hmm? We don't build with pcc. We build with either clang-8.0.1 or gcc-8.3.0, depending on platform.

Incidentally, however, pcc does support the -std=c99 flag.

Reply:
Whoops, it might have been NetBSD. Regardless, I do not want to depend on a non-standard option.